runbot_language/models/runbot_build.py

```diff
@@ -0,0 +1,87 @@
+# -*- encoding: utf-8 -*-
+##############################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: moylop260@vauxoo.com
+############################################################################
+
+'''
+    This file is used to add the field lang in runbot.build and the function
+      that install and assign the language to the users in the instance
+      generated.
+'''
+
+from openerp import api, fields, models, tools
+import logging
+from openerp.addons.runbot.runbot import run
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotBuild(models.Model):
+
+    '''
+    Inherit class runbot_build to add field to select the language &
+      the function with a job
+      to install and assign the language to users if this is captured
+      too is added with an super the
+      function create to assign the language from repo in the builds.
+    '''
+
+    _inherit = "runbot.build"
+
+    lang = fields.Selection(
+        tools.scan_languages(), 'Language',
+        help='Language to change '
+        'instance after of run test.', copy=True)
+
+    @api.multi
+    def cmd(self):
+        """Return a list describing the command to start the build"""
+        cmd, modules = super(RunbotBuild, self).cmd()
+        for build in self:
+            if build.lang and build.job == 'job_30_run':
+                cmd.append("--load-language=%s" % (build.lang))
+        return cmd, modules
+
+    @api.one
+    def update_lang(self):
+        """Set lang to all users into '-all' database"""
+        if self.lang:
+            db_name = "%s-all" % self.dest
+            try:
+                # update odoo version >=7.0
+                run(['psql', db_name, '-c', "UPDATE res_partner SET lang='%s' "
+                     "WHERE id IN (SELECT partner_id FROM res_users);" %
+                     (self.lang)])
+            except BaseException:
+                pass
```

Comment:
This have a exception because have twice compatibility
Between Version >=7.0 and version <7.0 `res_partner` model changed.
If you add a exception log you will have ever one log.